puppet/modules/jenkins_job_builder/files/theforeman.org/unmanaged_jobs.rb

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env ruby
+
+require 'net/http'
+require 'json'
+
+def help
+  puts "USAGE:\t#{__FILE__} JENKINS_JOB_BUILDER_INIFILE"
+end
+
+def managed?(job)
+  job['description']&.include?('<!-- Managed by Jenkins Job Builder -->')
+end
+
+def get_jenkins_url_from_ini(ini_file)
+  lines = File.open(ini_file, "r") do |file|
+    file.readlines
+  end
+
+  url_line = lines.select { |line| line.include?('url=') }&.first
+  url = url_line.gsub(/url=(.*\/\/)?/, '')&.chomp
```

Comment:
How does this work if the file doesn't include `url=` or `url =`? Does that mean `url_line` is `nil` and would `.gsub()` fail?

Reply:
`.gsub()` doesn't fail. it just doesn't replace anything.

```
[1] pry(main)> "".gsub(/url=(.*\/\/)?/, '')
""
```